### DISCUSSION TITLE
How to execute action in insert mode

### DISCUSSION THREAD

**Discussion by liuyehcf** [JetBrains/ideavim#946] [Q&A]
I'm using copilot, and I want to map action `copilot.applyInlays` to `π`(optional+p in mac) with following config:

```vim
inoremap π :action copilot.applyInlays<cr>
```

and it didn' work, because ideavim cannot execute actions in insert mode. And if I change it to

```vim
" <ecs> will change insert mode to normal mode
inoremap π <ecs>:action copilot.applyInlays<cr>
```

or

```vim
" <c-o> will change insert mode to normal mode
inoremap π <c-o>:action copilot.applyInlays<cr>
```

The action works fine, but the copilot suggestion dispear once trigger `<ecs>` or `<c-o>`, which change insert mode to normal mode. And finally nothing applied.

So, is there any way to config this map in `~/.ideavimrc` correctly？

**Comment by citizenmatt** [comment]
You can try the alternative `<Action>(…)` syntax, so something like `imap π <Action>(copilot.applyInlays)`. Note that this is `imap` and _not_ `inoremap` as it needs to map `<Action>(…)` to the action itself.
